test/e2e/tinkerbellhardwarecount.go

```diff
@@ -0,0 +1,43 @@
+package e2e
+
+import (
+	_ "embed"
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+)
+
+//go:embed tinkerbell_hardware_count.yaml
+var tinkerbellHardwareCountFile string
+
+// TinkerbellTest maps each Tinkbell test with the hardware count needed for the test.
+type TinkerbellTest struct {
+	Name  string `yaml:"name"`
+	Count int    `yaml:"count"`
+}
+
+// GetTinkerbellHardwareMap returns a map of tink tests with associated hardware count.
+func GetTinkerbellHardwareMap() (map[string]int, error) {
+	tinkerbellHardwareMap, err := readTinkerbellTestFile()
+	if err != nil {
+		return nil, fmt.Errorf("error getting hardware count for tinkerbell tests: %v", err)
+	}
+	return tinkerbellHardwareMap, nil
+}
+
+func readTinkerbellTestFile() (map[string]int, error) {
+	tinkerbellTests := make(map[string][]TinkerbellTest)
+	err := yaml.Unmarshal([]byte(tinkerbellHardwareCountFile), &tinkerbellTests)
+	if err != nil {
+		return nil, fmt.Errorf("unable to Unmarshal %s file: %v", TinkerbellHardwareCountFile, err)
```

Comment:
The error is a little misleading because this isn't a file, its embedded data.

```suggestion
		return nil, fmt.Errorf("unable to unmarshal hardware yaml: %v", TinkerbellHardwareCountFile, err)
```

I'm curious why we're embedding this to? It makes things quite rigid as configuration can't live separately from this code (i.e. is required at compile time).

Reply:
great question! Our E2E tests, when they are run we just copy the `e2e.test` binary to CodeBuild which means any additional files needed should be passed explicitly. since we are planning to establish this file as a requirement for Tinkerbell tests, embedding this avoid passing the file everytime explicitly (like we do presently for skipped_tests) to Codebuild during test runs.

Reply:
OK. I'm not sure I subscribe to the rationale but we can change later if needed.

Reply:
After talking this through and understanding a little more, it turns out the process is shared between all provider test runs so panicking would cause everything to fail. We'll continue returning the error.